### ISSUE TITLE
Keep GPS on while in c:geo and screen is off

### ISSUE THREAD

**Issue by andrixnet** [cgeo/cgeo#6549]
c:geo only uses GPS while running in foreground and with screen on. 
c:geo's purpose, by nature, is to provide location (GPS) based functionalities. For such a GPS focused application, to release the GPS in several, requiring restarting the GPS (even if it is a hot restart) and waiting for a fix, is not the best way towards users. 

Also as I have pointed out in other issues I have reported, please do not design c:geo in ways like expecting/requiring users to have the latest, most expensive, top of the line devices.
Such behaviour is detrimental to c:geo and geocaching penetration to potential users in (at least) some countries that are not among the most highly developed (economically). 
**And this means more then _a few with low end devices_.**

Now to the technical points. 

I strongly disagree with the strategy described here: http://www.cgeo.org/faq.html#gps-powersafe or the resolution here https://github.com/cgeo/cgeo/issues/4879 . Let me explain:

1. requiring users to manage an external app to keep the GPS module on is by far **not** the best attitude to adopt towards users, for an app whose core functionality is based on GPS.
_It also adds complications and frustration to new or less IT experienced users._
2. many apps adopt a UI switch to turn on/off the GPS while using that app. Just look at apps like Locus, OruxMaps, Androzic, Pathaway, etc.
Keep GPS on until explicitely turned off by user, or application explicitely exited (see below).
**IMO c:geo should really consider this concept.**
3. time, position change and signal level since last turned on DO affect the hot start of any GPS. Time to fix may be significantly longer due to poor signal conditions (eg tall buildings, forest cover, valleys), as compared to maintaning an existing fix while signal level is decreasing.
4. don't assume user has very high performance device, even if most of the developer have.
_In fact I would suggest to the developers to routinely test both on high performance and mid-range (lower mid-range) devices to gauge the performance of c:geo in  a more realistic way._

**IMO, as a GPS oriented app, I would expect c:geo to perform at least the following way:**
- c:geo is open, has focus. `GPS on`. 
- c:geo is open, has focus, screen is off. `GPS on`. (by turning off the screen either manually or by timeout is by no means equal to the intention of turning off the GPS)

**c:geo running in background in the following scenarios:** 
- c:geo has focus and the user changes app that has focus (window manager)
- c:geo has focus and user goes to home screen
These may be considered `GPS on` optional, or based on setting. But still, GPS on if GPS turn on button in c:geo was used

**c:geo is closed by explicit user action:**
- c:geo has focus, from main screen, user presses "BACK". Exit the app. Optionally display exit confirmation.
- c:geo has focus. User chooses "Exit" menu option (to be implemented).
See many other dedicated GPS apps. 

Note: turning off GPS on some screens in the app is nonsense. I seriously doubt going through the settings menus for half a minute incurs such a significant energy saving. And going through the settings menu is by design a seldom used feature to begin with.
And me leaving c:geo open in the settings menu draining battery because GPS is on, now that is my fault and my responsability.

------------------------
Even as an experienced IT person, I still find it frustrating and hassling to having to rely on an external app and the associated management, just to maintain GPS fix while I go geocaching with c:geo.

**Comment by SammysHP** [comment]
Duplicate of #4879…

**Comment by andrixnet** [comment]
IMO the discussion here covers a broader and more in depth scope. 
However it is meant to request a serious reconsideration of the matter.

**Comment by Lineflyer** [comment]
Thanks for taking the effort to elaborate this in detail.

But @SammysHP is right, that we already decided not to cover this topic from within c:geo but recommend an app which offers this dedicated functionality.
For me this is still a valid compromise and seen as reasonable also for the users who contacted us with similar issue on support mail (which I remember only 2-3 users in the last years).

**Comment by andrixnet** [comment]
Out of the total number of downloads recorded by Google Play, how many users actually routinely use c:geo to search for caches? (let's call this `users`)
Out of # of `users`, how many **actually post an issue** here when they encounter a problem? 
How many live with the problem, how many find a workaround, or how many drop c:geo because of it?

**Comment by andrixnet** [comment]
Imagine `Word`, with a separate application to change the font-face of the text, because it's not really a badly needed dedicated functionality and it's a reasonable compromise to use a third party app for those who really need it...

**Comment by Lineflyer** [comment]
I get your point of course. 
Besides my feedback I gave above, it is also a matter of effort and available developer resource to do it. If you can come along with a contribution for that or any developer here can evaluate the effort and confirm it is easy and quick to implement, we could do it.

From my (not all all expert) knowledge, e.g. keeping on GPS in settings and other parts of c:geo while the app is running should be easy to realize, but keeping it active as long as c:geo is in background can be more complicated as up to now c:geo does not have a formal way of exiting the program. Also extensive changes might be needed to control the background bahviour. Only my guess, maybe a developer proofs it wrong...

**Comment by SammysHP** [comment]
The only reasonable thing we could do is to keep GPS on while c:geo is in the foreground and the display switched off. Everything else is just guessing. This would require an additional option in the configuration.

A compromise might be to keep the GPS enabled only for the compass activity with a persistent switch in the menu (or a button like compass/GPS).

**Comment by pstorch** [comment]
One possible solution would be to start a Background Service to constantly receive location updates.
But suche a service adds complexity to the app. It's life cycle needs to be managed and the user needs to be able to see that it's running and be able to stop it.

**Comment by mirabilos** [comment]
Additionally:

- the device may not have sufficient RAM to support running multiple programs in parallel
- the user may not know any other GPS application or wish to not use them (e.g. for me, the geocaching application is my primary source of map and navigation data and everything)

I’d be okay with the “dashed line” compromise, however.

**Comment by Lineflyer** [comment]
>   the device may not have sufficient RAM to support running multiple programs in parallel

What? What device and what apps might cause this, which are supported by us? 
We are in this case talking about c:geo and a minor app keeping GPS on.

>  the user may not know any other GPS application or wish to not use them (e.g. for me, the geocaching application is my primary source of map and navigation data and everything)

For the IMHO minority of users requesting this, we provide a link to a suitable app in our FAQ AFAIR.

> For added fun, even if some GPS tracker booster is active, cgeo won’t update its trail.

Sure, the trail is another topic.

**Comment by Lineflyer** [comment]
I am sorry, but I will close this issue here.
As explained above such implementation would have a bad relation of effort and usefulness and we already decided some time ago not to implement keep-alive for GPS.

@cgeo team:
Feel free te reopen if you think its still worth working on it or if there is a volunteer for implementing such a feature.
